First Example:

Find the SVD of the given matrix:  
C = $\begin{bmatrix} 3 & 2 & 2 \\ 2 & 3 & -2 \end{bmatrix}$

Using the two famous formulas for SVD we know:  

1. $C^T$ $C$ = $V$ $\sum^T$ $\sum$ $V^T$  
  
2. $C$ $V$ = $U$ $\sum$  

  
  
We start from the first equation:  
$C^T$ = $\begin{bmatrix} 3 & 2 \\ 2 & 3 \\ 2 & -2 \end{bmatrix}$  
  

A = $C^T$ $C$ = $\begin{bmatrix} 3 & 2 \\ 2 & 3 \\ 2 & -2 \end{bmatrix}$ $\begin{bmatrix} 3 & 2 & 2 \\ 2 & 3 & -2 \end{bmatrix}$ = $\begin{bmatrix} 13 & 12 & 2 \\ 12 & 13 & -2 \\ 2 & -2 & 8 \end{bmatrix}$  
  

Next step, we have to diagonalize this matrix:

In [71]:
#Diagonalizing the Matrix A
using LinearAlgebra

A = [13 12 2; 12 13 -2; 2 -2 8]
F = eigen(A)
V = F.vectors
D = Diagonal(F.values)

A_reconstructed = F.vectors * Diagonal(F.values) * inv(F.vectors)


3×3 Matrix{Float64}:
 13.0  12.0   2.0
 12.0  13.0  -2.0
  2.0  -2.0   8.0

Since this diagonal eigenvalues is equal to :
$D$ = $\sum^T$ $\sum$  
And we know $\sum$ is diagonal, the entries of $\sum$ will be the square root of entries of $D$

In [72]:
#Displaying Sigma Part of the SVD
Sigma = Diagonal(sqrt.(D))
println("Sigma =")
Sigma_cut = map(x -> abs(x) < 1e-7 ? 0.0 : x, Sigma)



Sigma =


3×3 Diagonal{Float64, Vector{Float64}}:
 0.0   ⋅    ⋅ 
  ⋅   3.0   ⋅ 
  ⋅    ⋅   5.0

In [73]:
#Displaying V Part of the SVD

println("V =")
V_cut = map(x -> abs(x) < 1e-10 ? 0.0 : x, V)

V =


3×3 Matrix{Float64}:
  0.666667   0.235702  0.707107
 -0.666667  -0.235702  0.707107
 -0.333333   0.942809  0.0

Now that we have the V matrix and &\sum& matrix, we use the seconde equation to find the last part of the SVD, U matrix.  
  
$C$ $V$ = $U$ $\sum$
  
$\begin{bmatrix} 3 & 2 & 2 \\ 2 & 3 & -2 \end{bmatrix}$ $\begin{bmatrix} 0.666667 &  0.235702 & 0.707107\\ -0.666667 & -0.235702 & 0.707107 \\ -0.333333 &  0.942809 & 0.0\end{bmatrix}$ = $U$ $\begin{bmatrix} 0 & 0 & 0 \\ 0 & 3 & 0 \\ 0 & 0 & 5\end{bmatrix}$


In [74]:
C = [3 2 2; 2 3 -2]
H = C * V_cut
H_cut = map(x -> abs(x) < 1e-10 ? 0.0 : x, H)

2×3 Matrix{Float64}:
 0.0   2.12132  3.53553
 0.0  -2.12132  3.53553

In [75]:
Sigma = [0 0 0; 0 3 0; 0 0 5]
U = zeros(2, 3)
for i in 1:2  
    U[i, 2] = H_cut[i, 2] / Sigma[2, 2]
    U[i, 3] = H_cut[i, 3] / Sigma[3, 3]
end
U[:, 1] .= 0.0
println("U = ")

display(U)

2×3 Matrix{Float64}:
 0.0   0.707107  0.707107
 0.0  -0.707107  0.707107

U = 


In [76]:
C_reconstructed = U * Sigma * transpose(V_cut)

2×3 Matrix{Float64}:
 3.0  2.0   2.0
 2.0  3.0  -2.0